src/OrleansCodeGenerator/Utilities/SyntaxFactoryExtensions.cs

```diff
@@ -51,6 +51,62 @@ internal static class SyntaxFactoryExtensions
         }
         
         /// <summary>
+        /// Returns <see cref="TypeSyntax"/> for the provided <paramref name="type"/>.
+        /// </summary>
+        /// <param name="type">
+        /// The type.
+        /// </param>
+        /// <param name="includeNamespace">
+        /// A value indicating whether or not to include the namespace name.
+        /// </param>
+        /// <param name="includeGenericParameters">
+        /// Whether or not to include the names of generic parameters in the result.
+        /// </param>
+        /// <returns>
+        /// <see cref="TypeSyntax"/> for the provided <paramref name="type"/>.
+        /// </returns>
+        public static TypeSyntax GetTypeSyntaxAsDynamic(
+            this Type type,
+            bool includeNamespace = true,
+            bool includeGenericParameters = true)
+        {
+            if (type == typeof(void))
+            {
+                return SyntaxFactory.PredefinedType(SyntaxFactory.Token(SyntaxKind.VoidKeyword));
+            }
+
+            var code = type.GetParseableName(
```

Comment:
It would be preferable to get the generic parameters directly rather than by using regex. GetParseableName uses reflection anyway.